frontend/app/src/connection/StaticConnection.tsx

```diff
@@ -0,0 +1,157 @@
+   * For a StaticConnection, used for StatusWidget
+   */
+  onConnectionStateChange: OnConnectionStateChange
+
+  /**
+   * Function called when we receive a new ForwardMsg
+   */
+  onMessage: OnMessage
+}
+
+// Fetches the static asset url from the config file
+export async function getStaticConfig(): Promise<string> {
+  const isLocalStoreAvailable = localStorageAvailable()
+  let staticAssetUrl = ""
+
+  // Pull static asset url from localStorage if available
+  if (isLocalStoreAvailable) {
+    const cachedConfig = window.localStorage.getItem("stStaticAssetUrl")
+    if (cachedConfig) {
+      staticAssetUrl = cachedConfig
+    }
+  }
+
+  // Otherwise, fetch url from config file
+  if (!staticAssetUrl) {
+    try {
+      const response = await fetch(STATIC_ASSET_CONFIG, {
+        signal: AbortSignal.timeout(5000),
+      })
+
+      if (!response.ok) {
+        logError("Failed to fetch static config url: ", response.status)
+      } else {
+        const config = await response.json()
+        staticAssetUrl = config.static_url ?? undefined
+
+        // Set in localStorage
+        if (isLocalStoreAvailable && staticAssetUrl) {
+          window.localStorage.setItem("stStaticAssetUrl", staticAssetUrl)
+        }
+      }
+    } catch (err) {
+      logError("Failed to fetch static config url:", err)
+    }
+  }
+
+  return staticAssetUrl
+}
+
+// Fetches FowardMsg protos from S3 for static streamlit apps
+// First, gets the location of the static assets from url in the config
+// Then fetches the protos from that location
+export async function getProtoResponse(
+  staticNotebookId: string
+): Promise<void | ArrayBuffer> {
+  const staticAssetURL = await getStaticConfig()
+
+  // Next, fetch the static app's protos (if we have a url)
+  if (staticAssetURL) {
+    const path = `${staticAssetURL}/${staticNotebookId}/protos.pb`
+    const response = await fetch(path, {
+      signal: AbortSignal.timeout(5000),
+    })
+
+    if (!response.ok) {
+      logError(
+        `Failed to fetch static app protos for id: ${staticNotebookId}`,
+        response.status
+      )
+    } else {
+      const data = await response.arrayBuffer()
+      return data
+    }
+  }
+
+  return
+}
+
+// Triggers fetch of static app assets and dispatches ForwardMsgs to be handled
+// by App.tsx's handleMessage, replicating the app
+export async function dispatchAppForwardMessages(
+  staticNotebookId: string,
+  onMessage: OnMessage
+): Promise<void> {
+  const arrayBuffer = await getProtoResponse(staticNotebookId)
+
+  if (!arrayBuffer) {
+    logError("Failed to retrieve static app protos")
+    return
+  }
+
+  // Transforms our arrayBuffer response into ForwardMsgList protos
+  const forwardMsgList = ForwardMsgList.decode(new Uint8Array(arrayBuffer))
+
+  // Dispatches each ForwardMsg to be handled by App.tsx's handleMessage
+  forwardMsgList.messages.forEach(msg => {
+    onMessage(msg)
+  })
+
+  return
+}
+
+export function StaticConnection({
```

Comment:
Ah, so this is not a class, which was a confusion.

I would either:
1. Make this a class somehow (can be dummy).
2. rename this to `establishStaticConnection` so it's clear it's a verb for a function.

I think I would make this a class with a similar interface as WebSocketConnection, but I could be open to not being that.

Reply:
Was my initial thought to be more parallel to `WebSocketConnection`, but figured we were trying to transition away from classes and didn't want to over-complicate. Opted to rename to `establishStaticConnection`.